src/test/java/nextstep/subway/unit/LineServiceTest.java

```diff
@@ -171,4 +176,84 @@ void deleteSection() {
         );
     }
 
+
+    @DisplayName("출발역으로 부터 도착역까지의 경로에 있는 역 목록 및 경로 구간의 거리 조회")
+    @Test
+    void findShortestPathBetweenStations() {
+        // given
+        Line shinbundangLine = new Line("신분당선", "#D31145", gangnamStation, yangjaeStation, 1);
+        lineRepository.save(shinbundangLine);
+
+        Station dogokStation = stationRepository.save(new Station("도곡역"));
+        Station suseoStation = stationRepository.save(new Station("수서역"));
+        Line line3 = new Line("3호선", "#82C341", yangjaeStation, dogokStation, 2);
+        line3.addSection(dogokStation, suseoStation, 4);
+        lineRepository.save(line3);
+
+        Station seolleungStation = stationRepository.save(new Station("선릉역"));
+        Line line2 = new Line("2호선", "#0052A4", gangnamStation, seolleungStation, 2);
+        lineRepository.save(line2);
+
+        Line bundangLine = new Line("분당선", "#82C341", seolleungStation, dogokStation, 2);
+        bundangLine.addSection(dogokStation, suseoStation, 4);
+
+        // when
+        PathResponse pathResponse = lineService.findShortestPathBetweenStations(gangnamStation.getId(),
+                suseoStation.getId());
+
+        // when
+        List<String> stationNames = pathResponse.getStations().stream()
+                .map(Station::getName)
+                .collect(Collectors.toList());
+        Assertions.assertAll(
+                () -> assertThat(stationNames).isEqualTo(List.of("강남역", "양재역", "도곡역", "수서역")),
+                () -> assertThat(pathResponse.getDistance()).isEqualTo(7)
+        );
+    }
+
+    @DisplayName("조회시 출발역과 도착역이 같은 겨우 예외가 발생한다")
```

Comment:
```suggestion
    @DisplayName("경로 조회시 출발역과 도착역이 같은 겨우 예외가 발생한다")
```

조금 더 명확하게 표현하면 좋을 것 같아요 😄